optout/src/main/java/gov/cms/ab2d/optout/OptOutProcessorImpl.java

```diff
@@ -0,0 +1,69 @@
+package gov.cms.ab2d.optout;
+
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+@Slf4j
+@Setter
+@Component
+public class OptOutProcessorImpl implements OptOutProcessor {
+
+    @Value("${s3.region}")
+    private String s3Region;
+
+    @Value("${s3.bucket}")
+    private String s3Bucket;
+
+    @Value("${s3.filename}")
+    private String s3Filename;
+
+    @Override
+    public void process() {
+
+        //set region
+        final Region region = Region.of(s3Region);
+
+
+        // build S3 client
+        final S3Client s3Client =  S3Client.builder().region(region).build();
+
+
+        // build GetObjectRequest
+        final GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                .bucket(s3Bucket)
+                .key(s3Filename)
+                .build();
```

Comment:
As discussed, these will need to go into a separate bean for easier mocking (next PR).

Reply:
Okay.  In one of the subsequent stories, when adding the logic for extracting the consent/opt-out information and storing into the table, I'll extract the s3 specific code into a gateway bean class as discussed.

Reply:
👍